feat(notifier): adds checkout buy stepper

itofarina opened the pull request:
## Description
This PR is oriented to add the mockup with logical functionality for the notifier checkout.

## Features
- Adds stepper to integrate events subscription creation with payment step
- Adds `CurrencySelect` component

## Known issues
- SC interaction and probably some refactor to submit all data will be done in a new PR.
- I'm a bit confused about the use of different payment options. More details [here](https://github.com/rsksmart/rif-marketplace-ui/pull/862/files#diff-37043e8cecffcfb79ac7529c58766752b76070faa21e87bc54a94f1996f4be16R52).
- `PriceSummary` component was created to be reused in the future for multi-fiat, it supports it but the previous code doesn't. More info [here](https://github.com/rsksmart/rif-marketplace-ui/pull/862/files#diff-233dd75d3c6daa13d1ed9807a0c77ad265da3b74b1770a85e81aa0e5d770b6d4R16).

## Demo

https://user-images.githubusercontent.com/8449567/119567922-49b69500-bd83-11eb-9403-c7ff662208bb.mov

itofarina committed:
feat(notifier): adds checkout buy stepper

itofarina committed:
feat(notifier): adds payment mockup structure

itofarina committed:
feat: adds currency select component

itofarina committed:
feat(notifier): adds support for different quotations per currency

itofarina committed:
feat: adds price summary component

itofarina committed:
refactor(expirationDate): adds ability to define css class outside

itofarina committed:
feat(notifier): PriceSummary component styles and better typings
- CheckoutPayment component styles

itofarina committed:
feat(notifier): integrates event into checkout stepper

itofarina commented on src/components/organisms/notifier/buy/EventsRegistrar.tsx:
```diff
@@ -0,0 +1,129 @@
+import React, { FC, useState } from 'react'
```
There's almost nothing new in this file, this was extracted from `src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx`

itofarina committed:
feat(notifier): disables next on stepper until events are added

vjgene commented on src/components/molecules/ExpirationDate.tsx:
```diff
@@ -38,15 +38,15 @@ const ExpirationDate: FC<Props> = ({
       </Grid>
       <Grid item>
-        { type === 'warning' && <WatchLaterIcon /> }
+        {type === 'warning' && <WatchLaterIcon />}
```
can we use a constant here for ex. type === NOTIFIER_MESSAGES.WARNING?

itofarina replied:
Hey, thanks for this suggestion!
I thought you were right but after looking closer I realized that `type` variable is typed as `export type SubscriptionExpirationType = 'normal' | 'warning' | 'blocked'`

So it can only be one of those types or the code won't compile.

vjgene reviewed (commented):
Nice work!

jurajpiar commented on src/components/organisms/notifier/buy/CheckoutPayment.tsx:
```diff
@@ -0,0 +1,114 @@
+      <GridRow spacing={3} alignItems="center">
+        <Grid {...colProps}>
+          <Typography>Select currency</Typography>
+          <CurrencySelect
+            options={currencyOptions}
+            value={selectedCurrency}
+            onChange={handleCurrencyChange}
+          />
```
The currency choice has been removed from the design by agreement. unfortunately, the designs nor the story has been updated to reflect this change. The reason for this is that the consumer makes the choice on the offers page.

itofarina replied:
That changes a lot the implementation, thank you for this!

itofarina replied:
Done

jurajpiar commented on src/components/molecules/PriceSummary.tsx:
```diff
@@ -0,0 +1,46 @@
+      <Typography component="div" align="center">
+        <Box>{title}</Box>
+        <Box
+          fontWeight="fontWeightMedium"
+          fontSize="h6.fontSize"
+          color="primary.main"
```
that's rather odd notation. where is this from? I think I'd rather these were defined with JSS.

itofarina replied:
Resolving as we already discussed this one. [Reference](https://material-ui.com/system/typography/#font-size)

jurajpiar commented on src/components/organisms/notifier/buy/CheckoutStepper.tsx:
```diff
@@ -0,0 +1,87 @@
+          <CheckoutPayment
+            onBuy={onBuy}
+            paymentOptions={paymentOptions}
+            fiatUnit={SYSTEM_SUPPORTED_FIAT.usd}
```
use `exchangeRates.currentFiat` from the market context

itofarina replied:
Done, thank you!

jurajpiar commented on src/components/molecules/PriceSummary.tsx:
```diff
@@ -0,0 +1,46 @@
+  cryptoXRs: MarketCryptoRecord
+  // to fully support multifiat we need to refactor
+  // cryptoXR should be able to provide rate per crypto per fiat
+  fiatUnit: SYSTEM_SUPPORTED_FIAT
```
```suggestion
  cryptoPrice: Big
  currency: TokenXR
  fiat: BaseFiat
```

itofarina replied:
Done, thank you!

jurajpiar commented on src/components/organisms/notifier/buy/CheckoutPayment.tsx:
```diff
@@ -0,0 +1,114 @@
+  paymentOptions: QuotationPerToken
+  fiatUnit: SYSTEM_SUPPORTED_FIAT
+  expirationDate: Date
+  cryptoXRs: MarketCryptoRecord
```
```suggestion
  fiat: BaseFiat
  expirationDate: Date
  currency: TokenXR
```

itofarina replied:
Done, thank you!

jurajpiar commented on src/components/organisms/notifier/buy/CheckoutPayment.tsx:
```diff
@@ -0,0 +1,114 @@
+          </RoundedCard>
+          <GridRow justify="center">
+            <Typography color="secondary" align="center">Expiration date</Typography>
+            <ExpirationDate
+              className={classes.expirationDate}
+              type="normal"
+              date={expirationDate}
+            />
```
not sure it is that much improvement on:
```suggestion
            <Typography variant="body2" color="textSecondary">
              {getShortDateString(expirationDate)}
            </Typography>
```
considering that the icon is not going to be used here. Seems unnecessarily complicated to just display text.

itofarina replied:
Resolving after our discussion and agreement on reusing an existing component.

jurajpiar reviewed (changes requested):
Please remove the price selector and and use the currentFiat from marktetcontext as we discussed.

itofarina committed:
refactor(notifier-checkout): - removes currency select from last step
- simplifies price handling through the components using TokenXR
- removes QuotationPerToken type

jurajpiar committed:
Merge branch 'develop' into feature/notifierCheckout

jurajpiar committed:
refactor(notifier): makes use of getFiatPrice

jurajpiar committed:
refactor(notifier): renames checkout page
